### PR TITLE
Misc Vector updates

### PR DESCRIPTION
**Pull request by cgzones**
- Move swap() macro to source file and implement as function
- Implement Vector_get() and Vector_size() as inline functions
  to make them type safe and avoid lhs usage
- Comment comparison statistics, they are only needed for performance
  testing
- Use malloc, the code does not assume zero initialization and we might
  later use realloc() on the data

### CODE REVIEW

File: Vector.c [hunk 1]
```diff
@@ -18,9 +18,10 @@ Vector* Vector_new(const ObjectClass* type, bool owner, int size) {
 
    if (size == DEFAULT_SIZE)
       size = 10;
+   assert(size > 0);
    this = xMalloc(sizeof(Vector));
    this->growthRate = size;
-   this->array = (Object**) xCalloc(size, sizeof(Object*));
+   this->array = xMalloc(size * sizeof(Object*));
```
Comment (BenBE) [inline Vector.c:24]: Please overflow-check this multiplication.

File: Vector.c [hunk 2]
```diff
@@ -87,14 +88,22 @@ void Vector_prune(Vector* this) {
    this->items = 0;
 }
 
-static int comparisons = 0;
+//static int comparisons = 0;
+
+static void swap(Object** array, int indexA, int indexB) {
```
Comment (BenBE) [inline Vector.c:93]: Could as well be marked `inline` …

**Review by BenBE — CHANGES_REQUESTED** [review]

File: Vector.c (hunk 1 above)
Comment (cgzones) [reply Vector.c]: dropped this change; it seems not worth it after all

File: Vector.c (hunk 2 above)
Comment (cgzones) [reply Vector.c:93]: Optimizers should do the *right* thing on static/file-local functions

**Pull request merged**